Libraries/LibThread/Thread.cpp

```diff
@@ -62,22 +62,13 @@ void LibThread::Thread::start()
         rc = pthread_setname_np(m_tid, m_thread_name.characters());
         ASSERT(rc == 0);
     }
-    dbg() << "Started a thread, tid = " << m_tid;
+    dbgln("Started thread \"{}\", tid = {}", m_thread_name, m_tid);
 }
 
-void LibThread::Thread::join()
+int LibThread::Thread::join(void* thread_return)
 {
-    int rc = pthread_join(m_tid, nullptr);
+    int rc = pthread_join(m_tid, thread_return ? &thread_return : nullptr);
```

Comment:
This looks wrong. You probably want `void** thread_return` as the `Thread::join` argument and pass it through.

Reply:
Oh duh yeah, otherwise it just writes it to the local varaible.

Reply:
Nice catch!